node/src/effect.rs

```diff
@@ -1491,20 +1491,33 @@ impl<REv> EffectBuilder<REv> {
         .await
     }
 
-    /// Collects the switch blocks from the eras identified by provided era IDs. Returns
+    /// Collects the switch blocks for the eras identified by provided era IDs. Returns
     /// `Some(HashMap(era_id → block_header))` if all the blocks have been read correctly, and
-    /// `None` if at least one was missing.
+    /// `None` if at least one was missing. The header at EraId `n` is from the switch block of
+    /// era `n-1`, ie. it contains the data necessary for initialization of era `n`.
     pub(crate) async fn collect_switch_blocks<I: IntoIterator<Item = EraId>>(
         self,
         era_ids: I,
     ) -> Option<HashMap<EraId, BlockHeader>>
     where
         REv: From<StorageRequest>,
     {
-        futures::future::join_all(era_ids.into_iter().map(|era_id| {
-            self.get_switch_block_at_era_id_from_storage(era_id)
-                .map(move |maybe_block| maybe_block.map(|block| (era_id, block.take_header())))
-        }))
+        futures::future::join_all(
+            era_ids
+                .into_iter()
+                .filter_map(|era_id| {
+                    era_id
+                        .0
+                        .checked_sub(1)
+                        .map(|prev_era_num| (EraId(prev_era_num), era_id))
+                })
+                .map(|(prev_era_id, era_id)| {
+                    self.get_switch_block_at_era_id_from_storage(prev_era_id)
+                        .map(move |maybe_block| {
+                            maybe_block.map(|block| (era_id, block.take_header()))
+                        })
+                }),
```

Comment:
Maybe instead of subtracting 1 from the era ids, we should rename it to `get_switch_block_for_era_id` and the reasoning would be that switch block for `era-n` is in `era-{n-1}`.